backend/src/main/java/de/bund/digitalservice/ris/norms/domain/entity/CharacterRange.java

```diff
@@ -54,6 +59,121 @@ public boolean isValidCharacterRange() {
     return pattern.matcher(characterRange()).matches() && isEndGreaterStart();
   }
 
+  /**
+   * Finds the text-content of the range within the given node.
+   *
+   * @param node the node in which the range should be found
+   * @return the text-content of the range
+   */
+  public String findTextInNode(Node node) {
+    var text =
+        getNodesInRange(node).stream().map(Node::getTextContent).collect(Collectors.joining());
+    node.normalize();
+    return text;
+  }
+
+  /**
+   * Splits the given text node into multiple text nodes and returns the text node that includes the
+   * range from start to end of the text content.
+   *
+   * <p>A call to Node::normalize undoes this.
+   *
+   * @param textNode A text node
+   * @param start start of the text range within the node
+   * @param end end (excluding) of the text range within the node
+   * @return a text node that represents exactly the range
+   */
+  private Node extractTextNodeForRange(Node textNode, int start, int end) {
+    final var textContent = textNode.getTextContent();
+
+    // text before selection
+    if (start != 0) {
+      textNode
+          .getParentNode()
+          .insertBefore(
+              textNode.getOwnerDocument().createTextNode(textContent.substring(0, start)),
+              textNode);
+    }
+
+    // selected text
+    final var rangeTextContent = textContent.substring(start, Math.min(textContent.length(), end));
+    final var rangeNode =
+        textNode
+            .getParentNode()
+            .insertBefore(textNode.getOwnerDocument().createTextNode(rangeTextContent), textNode);
+
+    // text after selection
+    if (end < textContent.length()) {
+      textNode
+          .getParentNode()
+          .insertBefore(
+              textNode.getOwnerDocument().createTextNode(textContent.substring(end)), textNode);
+    }
+
+    textNode.getParentNode().removeChild(textNode);
+
+    return rangeNode;
+  }
+
+  /**
+   * Finds the nodes that make up the given range within the given node.
+   *
+   * <p>SIDE-EFFECT: This might split text-nodes into multiple text nodes to be able to return
+   * exactly the range. Please call Node::normalize after you are finished working with the nodes.
+   *
+   * @param node the node in which the range should be found
+   * @return a list of nodes that represent the given range
+   */
+  public List<Node> getNodesInRange(Node node) {
```

Comment:
Does this method need to be public? As far as I can see the entry point for this functionality is the other public method `findTextInNode`?

Reply:
Yes at the moment only that method uses this. But my idea is to use this method to figure out which parts of the xml need to be changed when applying a quoted text change command. Similar to this example from the test for this method: https://github.com/digitalservicebund/ris-norms/blob/35afbf26d92ffcde5c7b314732b2044383755be6/backend/src/test/java/de/bund/digitalservice/ris/norms/domain/entity/CharacterRangeTest.java#L227-L231. And for that this method needs then needs to be public.